src/systems/physics/Physics.cc

```diff
@@ -1700,6 +1705,75 @@ void PhysicsPrivate::CreateJointEntities(const EntityComponentManager &_ecm,
         {
           // No need to create this joint because it was already created when
           // parsing the model.
+
+          // Check if mimic constraint should be applied to this joint's axes.
+          auto jointAxis = _ecm.Component<components::JointAxis>(_entity);
+          auto jointAxis2 = _ecm.Component<components::JointAxis2>(_entity);
+
+          if (jointAxis)
+          {
+            if (auto mimic = jointAxis->Data().Mimic())
+            {
+              auto jointPtrMimic = this->entityJointMap
+                  .EntityCast<MimicConstraintJointFeatureList>(existingJoint);
+              if (jointPtrMimic)
+              {
+                jointPtrMimic->SetMimicConstraint(0,
+                    mimic->Joint(),
+                    mimic->Axis(),
+                    mimic->Multiplier(),
+                    mimic->Offset(),
+                    mimic->Reference());
+              }
+              else
+              {
+                static bool informed{false};
+                if (!informed)
+                {
+                  gzerr << "Attempting to create a mimic constraint for joint ["
+                        <<_name->Data()
+                        << "] but the chosen physics engine does not support "
+                        << "mimic constraints, so no constraint will be "
+                        << "created."
+                        << std::endl;
+                  informed = true;
+                }
+              }
+            }
+          }
+
+          if (jointAxis2)
+          {
+            if (auto mimic = jointAxis2->Data().Mimic())
+            {
+              auto jointPtrMimic = this->entityJointMap
+                  .EntityCast<MimicConstraintJointFeatureList>(existingJoint);
+              if (jointPtrMimic)
+              {
+                jointPtrMimic->SetMimicConstraint(1,
+                    mimic->Joint(),
+                    mimic->Axis(),
+                    mimic->Multiplier(),
+                    mimic->Offset(),
+                    mimic->Reference());
+              }
+              else
+              {
+                static bool informed{false};
+                if (!informed)
+                {
+                  gzerr << "Attempting to create a mimic constraint for joint ["
+                        <<_name->Data()
+                        << "] but the chosen physics engine does not support "
+                        << "mimic constraints, so no constraint will be "
+                        << "created."
+                        << std::endl;
+                  informed = true;
+                }
+              }
+            }
+          }
```

Comment:
these two code blocks are almost identical; a lambda function could potentially deduplicate this

Reply:
I deduplicated these code blocks in https://github.com/gazebosim/gz-sim/pull/1838/commits/1ccabce8aad6f0c44b231f22187708170e8e4aca, but it still doesn't work as discussed in https://github.com/gazebosim/gz-sim/pull/1838#discussion_r1290510976

Reply:
fixed in https://github.com/gazebosim/gz-sim/pull/1838/commits/e1448b451d1c5f82e1ce8d0d1710a3e57eb00efd